src/interface/guide/components/CastSequence.tsx

```diff
@@ -0,0 +1,169 @@
+import { useState } from 'react';
+import type Spell from 'common/SPELLS/Spell';
+import styled from '@emotion/styled';
+import { Tooltip } from 'interface';
+import { qualitativePerformanceToColor } from 'interface/guide';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import {
+  SectionContainer,
+  SectionHeader,
+  TitleColumn,
+  SectionTitle,
+  Label,
+  NavigationButtons,
+  NavButton,
+  NavCounter,
+  HelperText,
+  HelperTextRow,
+  ScrollableContainer,
+} from './GuideDivs';
+
+export interface CastInSequence {
+  timestamp: number;
+  spellId: number;
+  spellName: string;
+  icon: string;
+  performance?: QualitativePerformance;
+  tooltip?: React.ReactNode;
+}
+
+export interface CastSequenceEntry<T = unknown> {
+  data: T;
+  casts: CastInSequence[];
+  start?: number;
+  end?: number;
+}
+
+interface CastSequenceProps<T = unknown> {
+  spell: Spell;
+  sequences: CastSequenceEntry<T>[];
+  description?: string;
+  castTimestamp: (data: T) => string;
+  iconSize?: number;
+}
+
+/**
+ * Navigable cast sequence visualization showing spell icons in a filmstrip layout.
+ * @param spell - The spell/ability being analyzed
+ * @param sequences - Array of cast sequence entries containing cast data
+ * @param description - Optional description text displayed below the title
+ * @param castTimestamp - Function to format timestamp from cast data
+ * @param iconSize - Size in pixels for spell icons (default: 40)
+ */
+export default function CastSequence<T>({
+  spell,
+  sequences,
+  description,
+  castTimestamp,
+  iconSize = 40,
+}: CastSequenceProps<T>) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  if (!sequences || sequences.length === 0) {
+    return <div>No cast sequences to display</div>;
+  }
+
+  const currentSequence = sequences[currentIndex];
+
+  // Calculate window duration from start/end
+  let windowStart = currentSequence.start;
+  let windowEnd = currentSequence.end;
+
+  if (windowStart === undefined || windowEnd === undefined) {
+    // Fallback: calculate from casts
+    if (currentSequence.casts.length > 0) {
+      const timestamps = currentSequence.casts.map((c) => c.timestamp);
+      windowStart = Math.min(...timestamps);
+      windowEnd = Math.max(...timestamps);
+    }
+  }
+
+  const handlePrevious = () => {
+    setCurrentIndex((prev) => (prev > 0 ? prev - 1 : sequences.length - 1));
+  };
+
+  const handleNext = () => {
+    setCurrentIndex((prev) => (prev < sequences.length - 1 ? prev + 1 : 0));
+  };
```

Comment:
```suggestion
  const handlePrevious = useCallback(() => {
    setCurrentIndex((prev) => (prev > 0 ? prev - 1 : sequences.length - 1));
  }, [sequences.length]);

  const handleNext = useCallback(() => {
    setCurrentIndex((prev) => (prev < sequences.length - 1 ? prev + 1 : 0));
  }, [sequences.length]);
```

to prevent spurious rerenders